homeassistant/components/plaato/sensor.py

```diff
@@ -31,134 +23,62 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     """Set up the Plaato sensor."""
 
 
-async def async_setup_entry(hass, config_entry, async_add_entities):
+async def async_setup_entry(hass, entry, async_add_devices):
     """Set up Plaato from a config entry."""
     devices = {}
 
-    def get_device(device_id):
-        """Get a device."""
-        return hass.data[PLAATO_DOMAIN].get(device_id, False)
-
-    def get_device_sensors(device_id):
-        """Get device sensors."""
-        return hass.data[PLAATO_DOMAIN].get(device_id).get(PLAATO_DEVICE_SENSORS)
-
-    async def _update_sensor(device_id):
+    async def _update_sensor(device_id, sensor_data: PlaatoDevice):
         """Update/Create the sensors."""
-        if device_id not in devices and get_device(device_id):
-            entities = []
-            sensors = get_device_sensors(device_id)
-
-            for sensor_type in sensors:
-                entities.append(PlaatoSensor(device_id, sensor_type))
-
-            devices[device_id] = entities
-
-            async_add_entities(entities, True)
+        entry_data = hass.data[DOMAIN][entry.entry_id]
+        entry_data[SENSOR_DATA] = sensor_data
+
+        if entry.entry_id not in devices:
+            entry_data[DEVICE][DEVICE_ID] = device_id
+
+            entities = [
+                PlaatoSensor(hass.data[DOMAIN][entry.entry_id], sensor_type)
+                for sensor_type in sensor_data.sensors.keys()
+            ]
+            devices[entry.entry_id] = entities
+            async_add_devices(entities)
         else:
-            for entity in devices[device_id]:
-                async_dispatcher_send(hass, f"{PLAATO_DOMAIN}_{entity.unique_id}")
-
-    hass.data[SENSOR_DATA_KEY] = async_dispatcher_connect(
-        hass, SENSOR_UPDATE, _update_sensor
-    )
+            for entity in devices[entry.entry_id]:
+                async_dispatcher_send(hass, f"{DOMAIN}_{entity.unique_id}")
+
+    if entry.data.get(CONF_USE_WEBHOOK, False):
+        async_dispatcher_connect(hass, SENSOR_UPDATE, _update_sensor)
+    else:
+        coordinator = hass.data[DOMAIN][entry.entry_id][COORDINATOR]
+        if coordinator.data is not None:
+            async_add_devices(
```

Comment:
```suggestion
            async_add_entities(
```

These are entities not devices.